external_file_location/models/location.py

```diff
@@ -66,3 +67,7 @@ def onchange_protocol(self):
                 self.hide_port = True
             else:
                 self.hide_port = False
+            if cls._hide_address:
+                self.hide_address = True
+            else:
+                self.hide_address = False
```

Comment:
Well, this is a very sad implementation, relieved that it was not your idea haha. Thought about it a little, I think the correct implementation would be to make `hide_login` etc, as well as `port`, into computed fields, that all resolve to a single function `_compute_protocol_dependent_fields` that has an `@api.depends` clause on protocol. And then the function can just be the same as it is here.